flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkRewriteSubQueryRule.java

```diff
@@ -0,0 +1,195 @@
+ * }</pre>
+ */
+@Value.Enclosing
+public class FlinkRewriteSubQueryRule
+        extends RelRule<FlinkRewriteSubQueryRule.FlinkRewriteSubQueryRuleConfig> {
+
+    public static final FlinkRewriteSubQueryRule FILTER =
+            FlinkRewriteSubQueryRuleConfig.DEFAULT.toRule();
+
+    protected FlinkRewriteSubQueryRule(
+            FlinkRewriteSubQueryRule.FlinkRewriteSubQueryRuleConfig config) {
+        super(config);
+    }
+
+    @Override
+    public void onMatch(RelOptRuleCall call) {
+        Filter filter = call.rel(0);
+        RexNode condition = filter.getCondition();
+        RexNode newCondition = rewriteScalarQuery(condition);
+        if (condition.equals(newCondition)) {
+            return;
+        }
+
+        Filter newFilter = filter.copy(filter.getTraitSet(), filter.getInput(), newCondition);
+        call.transformTo(newFilter);
+    }
+
+    // scalar query like: `(select count(*) from T) > 0` can be converted to `exists(select * from
+    // T)`
+    private RexNode rewriteScalarQuery(RexNode condition) {
+        return condition.accept(
+                new RexShuttle() {
+                    public RexNode visitCall(RexCall call) {
+                        Optional<RexSubQuery> subQuery = getSupportedScalarQuery(call);
+                        if (subQuery.isPresent()) {
+                            return RexSubQuery.exists(subQuery.get().rel.getInput(0));
+                        }
+                        return super.visitCall(call);
+                    }
+                });
+    }
+
+    private boolean isScalarQuery(RexNode n) {
+        return n.isA(SqlKind.SCALAR_QUERY);
+    }
+
+    private boolean isBetween0And1(RexNode n, boolean include0, boolean include1) {
+        if (n instanceof RexLiteral) {
+            RexLiteral l = (RexLiteral) n;
+            if (l.getTypeName().getFamily() == SqlTypeFamily.NUMERIC && l.getValue() != null) {
+                double v = Double.parseDouble(l.getValue().toString());
+                return (0.0 < v && v < 1.0) || (include0 && v == 0.0) || (include1 && v == 1.0);
+            }
+        }
+        return false;
+    }
+
+    private boolean isCountStarAggWithoutGroupBy(RelNode n) {
+        if (n instanceof Aggregate) {
+            Aggregate agg = (Aggregate) n;
+            if (agg.getGroupCount() == 0 && agg.getAggCallList().size() == 1) {
+                AggregateCall aggCall = agg.getAggCallList().get(0);
+                return !aggCall.isDistinct()
+                        && aggCall.filterArg < 0
+                        && aggCall.getArgList().isEmpty()
+                        && aggCall.getAggregation() instanceof SqlCountAggFunction;
+            }
+        }
+        return false;
+    }
+
+    Optional<RexSubQuery> getSupportedScalarQuery(RexCall call) {
+        // check the RexNode is a RexLiteral which's value is between 0 and 1
+
+        // check the RelNode is a Aggregate which has only count aggregate call with empty args
+
```

Comment:
remove the comments, because it adds 
// check the RexNode is a RexLiteral which's value is between 0 and 1

        // check the RelNode is a Aggregate which has only count aggregate call with empty args